public/app/features/explore/Logs/Logs.tsx

```diff
@@ -271,6 +282,21 @@ const UnthemedLogs: React.FunctionComponent<Props> = (props: Props) => {
     }
   }, [logsVolumeData?.data, unregisterAllChildren, logsVolumeEnabled, hiddenLogLevels, register, toggleLegendRef]);
 
+  useEffect(() => {
+    if (getPinnedLogsCount() === PINNED_LOGS_LIMIT) {
+      setPinLineButtonTooltipTitle(
+        <span style={{ display: 'flex', textAlign: 'center' }}>
+          ❗️
+          <Trans i18nKey="explore.logs.maximum-pinned-logs">
+            Maximum of {{ PINNED_LOGS_LIMIT }} pinned logs reached. Unpin a log to add another.
+          </Trans>
+        </span>
+      );
```

Comment:
I know this was just moved here, but I think the UI is a bit weird:
![image](https://github.com/grafana/grafana/assets/8092184/74f647ad-f381-4105-b99a-fc81b9e8be91)

Reply:
Yeah I agree, a button with an icon that contains the tooltip or something else would be better then tooltipping the button.

Reply:
Let's leave this comment open and see what the explore folks say? I'd suggest we deal with this in a follow-up PR and ask UX to help us clean this up

Reply:
SGTM!

Reply:
This was a temporary design that @diegoadams proposed, and he can help with an improved design. @gtk-grafana you are suggesting having maybe some type of a warning icon as a button and not including it in the tooltip?

Reply:
Yeah but the `DataLinkButton` doesn't support passing in an Icon component yet, only icon as a string. Ideally we could house the tooltip within the icon in the button.